d2layouts/d2grid/layout.go

```diff
@@ -126,6 +122,34 @@ func withoutGridDiagrams(ctx context.Context, g *d2graph.Graph, layout d2graph.L
 			}
 			if obj.LabelDimensions.Height != 0 {
 				labelHeight := float64(obj.LabelDimensions.Height) + 2*label.PADDING
+
+				{
+					// also check for grid cells with outside top labels
+					topY := gd.objects[0].TopLeft.Y
+					highestLabel := topY
+					for _, o := range gd.objects {
+						if o.TopLeft.Y > topY {
+							if gd.rowDirected {
```

Comment:
why is this in here?

Reply:
I'll add a comment, agree it isn't obvious